### ISSUE TITLE
Add option to convert on save

### ISSUE THREAD

**Issue by onetrev** [hilleer/vscode-yaml-plus-json#157]
I was thinking it would be very handy to have an option to convert current YAML file on save. I'm guessing this could help others too? My use case is I'm using this extension because working with `theme.json` to develop a large custom WP theme is a nightmare. It's much easier in YAML.

I already have `keepOriginalFiles` and `overwriteExistentFiles` as **`always`**. I'm thinking that if the above feature is added these two should be always by default then since that would be the natural work flow right? Thanks!

**Comment by hilleer** [comment]
Hey @onetrev 

Thanks for the feature request! 🥳 

Just to get it right: So flow would be like this:

1. Set an option/config value e.g. `convertOnSave: true`
2. Create a file - yaml or json
3. write something
4. Save file - automatically converts to the counterpart (yaml --> json, json --> yaml)

**Comment by hilleer** [comment]
Cool! I will consider it an see what is possible.

What if you do it like described above but additionally:

5. Make some more changes to source file.
6. Save file - automatically convert. Does it overwrite the (already) existent file automatically? Would it warn about it? 👀

**Comment by onetrev** [comment]
Awesome, thanks for considering this!

Good questions. I think _safety first_, it should warn you about overwriting the file, unless yes you have `overwriteExistentFiles` as always. Good idea!

**Comment by gyurmey2** [comment]
It would be great!

**Comment by mgawe** [comment]
Hi @hilleer, is there any chance to implement this?

**Comment by hilleer** [comment]
> Hi @hilleer, is there any chance to implement this?

Definitely. I have just been busy. PRs are also welcome but otherwise I'll try to find some time soon 🙂 🤞

**Comment by mgawe** [comment]
I haven't found a better VSC extension (YAML to JSON), and this would just be icing on the cake. Thanks!

**Comment by gyurmey2** [comment]
Hi @hilleer, I just wanted to ask if this is even planned? And I just want to make sure ;)

**Comment by hilleer** [comment]
> Hi @hilleer, I just wanted to ask if this is even planned? And I just want to make sure ;)

Yes. But I haven't had time to do it unfortunately 🕐 Pull requests are more than welcomed.
